src/_pytest/_code/source.py

```diff
@@ -48,7 +55,15 @@ def __eq__(self, other):
     # Ignore type because of https://github.com/python/mypy/issues/4266.
     __hash__ = None  # type: ignore
 
-    def __getitem__(self, key):
+    @overload
+    def __getitem__(self, key: int) -> str:
+        ...
+
+    @overload  # noqa: F811
+    def __getitem__(self, key: slice) -> "Source":
+        ...
```

Comment:
With regard to coverage you could use `raise NotImplementedError()` here (ignored via config), or we could also add "..." to be ignored by default.

Reply:
Thanks, changed to `raise NotImplementedError()`.